processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -454,41 +457,90 @@ func (c *WatchClient) addOrUpdatePod(pod *api_v1.Pod) {
 		}
 	}
 
+	return newPod
+}
+
+func (c *WatchClient) getIdentifiersFromAssoc(newPod *Pod) []PodIdentifier {
+	ids := []PodIdentifier{}
+	for _, assoc := range c.Associations {
+		ret := []string{}
+		for _, source := range assoc.Sources {
+			// If association configured to take IP address from connection
+			switch {
+			case source.From == "connection":
+				if newPod.Address != "" {
+					ret = append(ret, newPod.Address)
+				}
+			case source.From == "resource_attribute":
+				attr := ""
+				switch source.Name {
+				case conventions.AttributeK8SNamespaceName:
+					attr = newPod.Namespace
+				case conventions.AttributeK8SPodName:
+					attr = newPod.Name
+				case conventions.AttributeK8SPodUID:
+					attr = newPod.PodUID
+				case conventions.AttributeHostName:
+					attr = newPod.Address
+				default:
+					if v, ok := newPod.Attributes[source.Name]; ok {
+						attr = v
+					}
+				}
+
+				if attr != "" {
+					ret = append(ret, attr)
+				}
+			}
+		}
+
+		if len(ret) == len(assoc.Sources) {
+			ids = append(ids, PodIdentifier(strings.Join(ret, PodIdentifierDelimiter)))
```

Comment:
I don't think we can consider this as a reliable identifier given that now we allow using value of any attribute.

Let's say I have two sets of sources in `pod_association`, one uses values from `k8s.pod.name` attribute, another one uses `container.image`. It's theoretically possible that I have two different pods having the same value of `k8s.pod.name` attribute from one pod and `container.image` from another. 

I think we should use an array of attribute key-value pairs as an identifier.

Reply:
Yeah, that's true, but as we have to keep also `connection` type to be consistent I propose the following structure:

```
// PodAttributeIdentifier identifies pod attribute value
type PodAttributeIdentifier struct {
	key   AssociationSource
	value string
}

// PodIdentifier is a custom type to represent Pod by list of PodAttributeIdentifier
type PodIdentifier []PodAttributeIdentifier
```

Reply:
I revisited my structure, and this will be better:
```
// PodIdentifier is a custom type to represent Pod by list of PodAttributeIdentifier
type PodIdentifier map[AssociationSource]string
```

Reply:
but it won't work, as `incomparable map key type PodIdentifiercompiler[IncomparableMapKey](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal?utm_source%3Dgopls#IncomparableMapKey)`
:confused:

Reply:
Array of structs should be a valid identifier. Something like this: `[maxSourcesLen]PodAssociationSourceConfig` where maxSourcesLen is max length across all the sources in `pod_association` list

Reply:
Sounds good. thanks